kube/src/client/mod.rs

```diff
@@ -98,22 +78,22 @@ impl Client {
     ///
     /// Will fail if neither configuration could be loaded.
     pub async fn infer() -> Result<Self> {
-        let configuration = crate::config::Configuration::infer().await?;
-
-        let client_builder = match crate::config::incluster_client() {
+        let mut client_config = match ClientConfig::infer().await {
             Ok(c) => c,
-            Err(_) => client_builder::create(ConfigOptions::default()).await?,
+            Err(_) => ClientConfig::new_from_config_file(Default::default()).await?,
         };
+        let base_path = std::mem::replace(&mut client_config.base_path, String::new());
```

Comment:
mem::replace seems pretty low level for this

Reply:
Removed